README.md

```diff
@@ -1,3 +1,92 @@
 # velocity-plugins-index
 
-This repository hosts the index for UrbanCode Velocity plugins that instances of UrbanCode velocity reach out to for plugin installs/updates.
+This repository hosts the index for UrbanCode Velocity plugins which instances of UrbanCode velocity reach out to for plugin installs/updates.
+
+## Repository Structure
+
+* `plugins`: A directory containing a list of directories of plugins by `pluginId`. Each plugin directory should have an `info.json` and `releases.json` file.
+
+  * `info.json`: A file containing a JSON object of general information about the plugin that will help users identify if they want to install the plugin to their instance of UrbanCode Velocity.
+  * `releases.json`: A file containing a JSON array of release objects, each release object contianing the information necessary to install the release and some optional notes to help users identify if they should upgrade to that plugin version in their instance of UrbanCode Velocity.
+
+* `index.json`: A file generated based on the `plugins` directory contents, contianing a JSON object with keys for each `pluginId` and values contianing the plugin general information and latest release information.
+
+* `src`: A directory contianing the scripts required to generate the `index.json` file, as well as lint that file and the `plugins` directory contents to ensure everything is in sync.
+
+## JSON Structures
+
+There are 2 main file types that must abide by strict guidelines:
+
+* `info.json`: A JSON object containing the following fields:
+
+    | Field | Type | Required | Description |
+    | ----- | ---- | -------- | ----------- |
+    | name | string | yes | The display name of the plugin. This is how users will identify the plugin in UrbanCode Velocity. |
+    | url | string | yes | A URL to a website containing information about the plugin. A user would use this information to find out more about the plugin. |
+    | description | string | yes | A short description about what the plugin does. This should provide enough context for a user to determine whether the plugin is right for their needs. |
+    | author | object | yes | Who is responsible for creating/updating the plugin. This will let a user know who is responsible for maintaning the plugin. |
```

Comment:
```suggestion
    | author | object | yes | Who is responsible for creating/updating the plugin. This will let a user know who is responsible for maintaining the plugin. |
```